include/find.hpp

```diff
@@ -18,14 +18,52 @@
 
 namespace bit {
 
-// TODO
 template <class InputIt, class T>
 constexpr bit_iterator<InputIt> find(bit_iterator<InputIt> first,
```

Comment:
So I don't think this will work with InputIt either. I think, though, that it can easily be modified to. Instead of iterating over a bit_iterator until `cursor != last`, you could create a word iterator `auto it = first.base()` and then iterate over it until `it == std::next(last.base(), !is_last_aligned)`, at which point you can use the same logic you have already been using.

From an optimization standpoint, I think the above works better anyways. (1) it avoids having to call `std::min`, `get_word` and a few other arithmetic operations on every iteration. As another side note, I think `get_word` should only be used when we potentially need more than 2 words. If we only need 2 words, it would be faster to just use `shld`/`shrd` I think, since its the same logic just without extra stuff.